lib/task.js

```diff
@@ -351,10 +435,64 @@ Task.prototype = {
       pullOptions.authconfig = credentials;
     }
 
-    return yield pullImage(
+    return await pullImage(
       this.runtime.docker, dockerImageName, this.stream, pullOptions
     );
-  },
+  }
+
+  isAborted() {
+    return this.taskState === 'aborted';
+  }
+
+  isCanceled() {
+    return this.taskState === 'canceled';
+  }
+
+  /**
+   * Aborts the running of the task.  This is similar to cancelling a task, but
+   * will allow time to upload artifacts and report the run as an exception instead.
+   *
+   * @param {String} reason - Reason for aborting the test run (Example: worker-shutdown)
+   */
+  abort(reason='aborted') {
+    this.taskState = 'aborted';
+    this.taskException = reason;
```

Comment:
Please explain the default string? It's not a valid _reason_ as far as I'm aware.
Valid exceptions that a worker can report is: `worker-shutdown` and `malformed-payload`.